llm/throughput/README.md

```diff
@@ -0,0 +1,165 @@
+# MosaicGPT Throughput, MFU, and HFU
+Throughput, MFU, and HFU for different sized MosaicGPT models trained on MCloud.
+MFU and HFU are defined in https://arxiv.org/abs/2205.05198
+All FLOP calculations do not include norm, act, and residual.
+
+MFU approximation:
+Using only the parameters, per token, each parameter is used for a MAC (2 FLOPS) per network operation. Neural Network training has 4 network operations: forward pass, backward pass, and computation of parameter gradient.
+The attention mechanism forward pass FLOPS are: `attn_flops_per_seq = n_layers * 2 * 2 * (d_model * (seq_len**2))`
+```
+flops_per_token = 2 * n_params
+flops_per_seq = flops_per_token * seq_len
+mfu* = 3 * flops_per_seq * throughput / (gpu_num * GPU_AVAILABLE_FLOPS)
+
+attn_flops_per_seq = n_layers * 2 * 2 * (d_model * (seq_len**2))
+mfu = (3 * flops_per_seq + 3 * attn_flops_per_seq) * throughput / (gpu_num * GPU_AVAILABLE_FLOPS)
+```
+
+HFU is approximated using the fact that the networks use checkpointing for activation recomputation. This effectively requires an extra forward pass through the network.
+```
+hfu* = 4 * flops_per_seq * throughput / (gpu_num * GPU_AVAILABLE_FLOPS)
+hfu = (4 * flops_per_seq + 4 * attn_flops_per_seq) * throughput / (gpu_num * GPU_AVAILABLE_FLOPS)
+```
+
+These are approximations. Real HFU would be higher since it would include the FLOP usage of norm, act, residual, as well as account for ALL recomputation (the model uses FlashAttn which already does a recomputation in the fwd pass; in reality, the hfu attn multiplier should be 5 instaed of 3).
+
+Below we highlight some of the configurations, but the full tables can be found in [FULL_TABLES](./FULL_TABLES.md)
+
+## A100 80GB
```

Comment:
The "highlights" are still quite long, do you think we should try to shorten them? Maybe like keeping the seqlen=[2048...16384] sweep but cutting down on the GPUType=[..] sweep? I'm not super opinionated here but it does feel like a lot even when looking on my laptop.

Reply:
cut down on a few